File: cdk-from-cfn-testing/expected/custom_resource/csharp/CustomResourceStack.cs
using Amazon.CDK;
using Amazon.CDK.AWS.IAM;
using Amazon.CDK.AWS.Lambda;
using Constructs;
using System.Collections.Generic;

namespace CustomResourceStack
{
    public class CustomResourceStackProps : StackProps
    {
    }

    /// <summary>
    /// Test Custom Resource conversion
    /// </summary>
    public class CustomResourceStack : Stack
    {
        public CustomResourceStack(Construct scope, string id, CustomResourceStackProps props = null) : base(scope, id, props)
        {

            // Resources
            var lambdaRole = new CfnRole(this, "LambdaRole", new CfnRoleProps
            {
                AssumeRolePolicyDocument = new Dictionary<string, object>
                {
                    { "Version", "2012-10-17"},
                    { "Statement", new []
                    {
                        new Dictionary<string, object>
                        {
                            { "Effect", "Allow"},
                            { "Principal", new Dictionary<string, object>
                            {
                                { "Service", "lambda.amazonaws.com"},
                            }},
                            { "Action", "sts:AssumeRole"},
                        },
                    }},
                },
            });
            var backingLambda = new CfnFunction(this, "BackingLambda", new CfnFunctionProps
            {
                Runtime = "python3.9",
                Handler = "index.handler",
                Role = lambdaRole.AttrArn,
                Code = new CfnFunction.CodeProperty
                {
                    ZipFile = @"def handler(event, context):
                      return {'Status': 'SUCCESS', 'Data': {'Endpoint': 'test-endpoint'}}
                    ",
                },
            });
            var myCustomResource = new CfnCustomResource(this, "MyCustomResource", new CfnCustomResourceProps
            {
                ServiceToken = backingLambda.AttrArn,
            });
            myCustomResource.AddOverride("Type", "Custom::DatabaseSetup");
            myCustomResource.AddPropertyOverride("DatabaseName", "mydb");
            myCustomResource.AddPropertyOverride("TableCount", 5);
            myCustomResource.AddPropertyOverride("EnableLogging", "true");
            myCustomResource.AddPropertyOverride("Tags", new []
            {
                "prod",
                "critical",
            });
            myCustomResource.CfnOptions.DeletionPolicy = CfnDeletionPolicy.RETAIN;
            myCustomResource.AddDependency(backingLambda);
            var consumerLambda = new CfnFunction(this, "ConsumerLambda", new CfnFunctionProps
            {
                Runtime = "python3.9",
                Handler = "index.handler",
                Role = lambdaRole.AttrArn,
                Code = new CfnFunction.CodeProperty
                {
                    ZipFile = @"def handler(event, context):
                      pass
                    ",
                },
                Environment = new CfnFunction.EnvironmentProperty
                {
                    Variables = new Dictionary<string, string>
                    {
                        { "DB_ENDPOINT", myCustomResource.GetAtt("Endpoint").ToString()},
                    },
                },
            });
        }
    }
}

File: cdk-from-cfn-testing/expected/custom_resource/typescript/CustomResourceStack.ts
import * as cdk from 'aws-cdk-lib';
import * as iam from 'aws-cdk-lib/aws-iam';
import * as lambda from 'aws-cdk-lib/aws-lambda';

export interface CustomResourceStackProps extends cdk.StackProps {
}

/**
 * Test Custom Resource conversion
 */
export class CustomResourceStack extends cdk.Stack {
  public constructor(scope: cdk.App, id: string, props: CustomResourceStackProps = {}) {
    super(scope, id, props);

    // Resources
    const lambdaRole = new iam.CfnRole(this, 'LambdaRole', {
      assumeRolePolicyDocument: {
        Version: '2012-10-17',
        Statement: [
          {
            Effect: 'Allow',
            Principal: {
              Service: 'lambda.amazonaws.com',
            },
            Action: 'sts:AssumeRole',
          },
        ],
      },
    });

    const backingLambda = new lambda.CfnFunction(this, 'BackingLambda', {
      runtime: 'python3.9',
      handler: 'index.handler',
      role: lambdaRole.attrArn,
      code: {
        zipFile: 'def handler(event, context):\n  return {\'Status\': \'SUCCESS\', \'Data\': {\'Endpoint\': \'test-endpoint\'}}\n',
      },
    });

    const myCustomResource = new cdk.CfnCustomResource(this, 'MyCustomResource', {
      serviceToken: backingLambda.attrArn,
    });
    myCustomResource.addOverride('Type', 'Custom::DatabaseSetup');
    myCustomResource.addPropertyOverride('DatabaseName', 'mydb');
    myCustomResource.addPropertyOverride('TableCount', 5);
    myCustomResource.addPropertyOverride('EnableLogging', 'true');
    myCustomResource.addPropertyOverride('Tags', [
      'prod',
      'critical',
    ]);
    myCustomResource.cfnOptions.deletionPolicy = cdk.CfnDeletionPolicy.RETAIN;
    myCustomResource.addDependency(backingLambda);

    const consumerLambda = new lambda.CfnFunction(this, 'ConsumerLambda', {
      runtime: 'python3.9',
      handler: 'index.handler',
      role: lambdaRole.attrArn,
      code: {
        zipFile: 'def handler(event, context):\n  pass\n',
      },
      environment: {
        variables: {
          'DB_ENDPOINT': myCustomResource.getAtt('Endpoint').toString(),
        },
      },
    });
  }
}
